theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -268,6 +270,10 @@ def _merge_infos(
             M & (grad_loop_info.status == NonlinearOptimizerStatus.MAX_ITERATIONS)
         ] = -1
 
+    def _update_tmp_optim_vars(self):
```

Comment:
Might be helpful to leave a brief comment. Are the tmp vars still in the compute graph?

Reply:
Added comments at the top of this method.